### PR TITLE
Add ability to blacklist entire databases

### PR DESCRIPTION
**Pull request by jpeyper**
Similar to the blacklist_tables option, this new setting will allow entire databases to be blacklisted.

This should address #260

### CODE REVIEW

**Comment by jpeyper** [comment]
@osheroff My first PR for you. Hopefully I've followed the you're protocols / styles properly :)

**Comment by jpeyper** [comment]
This should address #289

File: src/main/java/com/zendesk/maxwell/MaxwellTableCache.java [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package com.zendesk.maxwell;
 
 import java.util.HashMap;
+import java.util.HashSet;
```
Comment (nmaquet) [inline src/main/java/com/zendesk/maxwell/MaxwellTableCache.java]: Unused import?

**Comment by nmaquet** [comment]
Cool ! This will be useful :+1:

**Comment by osheroff** [comment]
lgtm, fix the unused import and I'll merge.  thanks!

**Comment by jpeyper** [comment]
@osheroff done.

File: src/main/java/com/zendesk/maxwell/MaxwellTableCache.java [hunk 2]
```diff
@@ -22,7 +23,7 @@ public void processEvent(Schema schema, MaxwellFilter filter, TableMapEvent even
 
 			Table tbl = db.findTable(tblName);
 
-			if ( filter != null && filter.isTableBlacklisted(tblName) )
+			if ( filter != null && filter.isTableBlacklisted(tbl.getDatabase().getName(), tblName) )
```
Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/MaxwellTableCache.java]: @jpeyper my eyes sort of glossed over this line, but in local testing I found that it's subject to crashing -- if the database is blacklisted, it's quite possible that `db.findTable()` will return `null` (if we didn't process the `CREATE TABLE` ddl), or the call to `schema.findDatabase` may even return null (if we didn't process the `CREATE DATABASE` ddl).  

Can you do an integration test that sets up a database blacklist and then runs through a few scenarios?  notably:

  - create DB
  - capture schema
  - blacklist DB
  - create table DB.tbl
  - insert into DB.tbl

and also

  - capture schema
  - blacklist DB
  - create DB
  - create tbl
  - insert into DB.tbl

Comment (jpeyper) [inline src/main/java/com/zendesk/maxwell/MaxwellTableCache.java]: I'm developing in windows, and I get errors when running you're test suite... but i did just notice that just above that line there is already a `dbName` variable in scope (on line 17).

So I've switched to just using that.

Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/MaxwellTableCache.java]: that will handle the case where the table has not been tracked inside the database, but not the case where the database has not been tracked

Comment (jpeyper) [inline src/main/java/com/zendesk/maxwell/MaxwellTableCache.java]: Ok, I think I see now.

Still not sure how to do the integration tests, but I think the change I've just made should cover the crash cases you've seen.

Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/MaxwellTableCache.java]: ok.  I'll add an integration test for the cases before I merge.

**Comment by jpeyper** [comment]
Was there anything else that needs addressing?

**Comment by osheroff** [comment]
nope, all good.  merged into master.

**Comment by osheroff** [comment]
(after adding a couple of tests)

**Comment by wushujames** [comment]
Congrats to @jpeyper on his first PR to Maxwell!

**Comment by jpeyper** [comment]
@osheroff Any idea when this will get into a release / RC?

**Comment by osheroff** [comment]
@jpeyper I'm trying to stabilize maxwell right now for a big-ol 1.0 release (just picking up bugfixes), but I'm happy to do a 1.1-PRE release with whatever's in master if that'd help.

**Comment by osheroff** [comment]
https://github.com/zendesk/maxwell/releases/tag/v1.1.0-PRE1

**Pull request merged**